web/src/message_events_util.ts

```diff
@@ -1,39 +1,59 @@
+import {z} from "zod";
+
 import * as blueslip from "./blueslip";
 import * as channel from "./channel";
 import * as compose_notifications from "./compose_notifications";
 import * as message_lists from "./message_lists";
+import type {MessageList, RenderInfo} from "./message_lists";
 import * as message_store from "./message_store";
+import type {Message} from "./message_store";
 import * as narrow_state from "./narrow_state";
 import * as unread_ops from "./unread_ops";
 import * as util from "./util";
 
-// TODO: Move this function to 'message_util.ts' once #30702 is merged.
```

Comment:
Removed this comment since we are planning to have this as a separate module. [discussion](4c8cee548ef3ca27a6617d5678f0cf591649f6bb)